src/test/java/usecases/get_story_comments/GscInteractorTests.java

```diff
@@ -0,0 +1,150 @@
+package usecases.get_story_comments;
+
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+import usecases.CommentRepoData;
+import usecases.RepoRes;
+import usecases.Response;
+import usecases.ThreadRegister;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+
+public class GscInteractorTests {
+    private static final ThreadRegister register = new ThreadRegister();
+    private static final CommentRepoData duplicateCRD = new CommentRepoData(1, 1, "Julius", "Wow, just wow");
+    private static final List<CommentRepoData> duplicateRow = new ArrayList<CommentRepoData>(List.of(duplicateCRD));
+    GscOutputBoundary pres;
+    GscGatewayComments repo;
+
+    /**
+     * Customizable class to imitate GscPresenter during testing
+     */
+    static class CustomizableGscOutputBoundary implements GscOutputBoundary {
+
+        private GscOutputData receivedData;
+
+        public CustomizableGscOutputBoundary() {
+            this.receivedData = null;
+        }
+
+        public void putStoryComments(GscOutputData d) {
+            this.receivedData = d;
+        }
+
+        public GscOutputData getReceivedData() {
+            return this.receivedData;
+        }
+
+        @Override
+        public void outputShutdownServer() {
+            throw new RuntimeException("This method is not implemented and should not be called");
+        }
+    }
+
+    /**
+     * Customizable class to imitate repository with stories during testing.
+     * Records received data to be compared with applied requests
+     */
+    static class CustomizableGscGatewayComments implements GscGatewayComments {
+
+        private final boolean getAllCommentsSuccess;
+
+        /**
+         * Constructor of our customizable gateway
+         * @param getAllCommentsSuccess determines return value for getAllComments
+         */
+        public CustomizableGscGatewayComments(boolean getAllCommentsSuccess) {
+            this.getAllCommentsSuccess = getAllCommentsSuccess;
+        }
+
+        /**
+         * Method imitating repository's getAllComments with custom element based on
+         * getAllCommentsSuccess parameter.
+         * @param storyId requested storyId
+         * @return "success" and predetermined output when commentAsGuestSuccess is true,
+         * and "failure" with null otherwise
```

Comment:
once again, better to write 'success or failure response'. i am assuming this method was originally supposed to return a string